src/zarr/core/array.py

```diff
@@ -3449,3 +3456,305 @@ def _get_default_codecs(
         dtype_key = "numeric"
 
     return [{"name": codec_id, "configuration": {}} for codec_id in default_codecs[dtype_key]]
+
+
+FiltersParam: TypeAlias = (
+    Iterable[dict[str, JSON] | Codec] | Iterable[numcodecs.abc.Codec] | Literal["auto"]
+)
+CompressionParam: TypeAlias = (
+    Iterable[dict[str, JSON] | Codec] | Codec | numcodecs.abc.Codec | Literal["auto"]
+)
+
+
+async def create_array(
+    store: str | StoreLike,
+    *,
+    name: str | None = None,
+    shape: ShapeLike,
+    dtype: npt.DTypeLike,
+    chunk_shape: ChunkCoords | Literal["auto"] = "auto",
+    shard_shape: ChunkCoords | None = None,
+    filters: FiltersParam = "auto",
+    compression: CompressionParam = "auto",
+    fill_value: Any | None = 0,
+    order: MemoryOrder | None = "C",
+    zarr_format: ZarrFormat | None = 3,
+    attributes: dict[str, JSON] | None = None,
+    chunk_key_encoding: ChunkKeyEncoding | ChunkKeyEncodingParams | None = None,
+    dimension_names: Iterable[str] | None = None,
+    storage_options: dict[str, Any] | None = None,
+    overwrite: bool = False,
+    config: ArrayConfig | ArrayConfigParams | None = None,
+    data: npt.ArrayLike | None = None,
+) -> AsyncArray[ArrayV2Metadata] | AsyncArray[ArrayV3Metadata]:
+    """Create an array.
+
+    Parameters
+    ----------
+    store : str or Store
+        Store or path to directory in file system or name of zip file.
+    name : str or None, optional
+        The name of the array within the store. If ``name`` is ``None``, the array will be located
+        at the root of the store.
+    shape : ChunkCoords
+        Shape of the array.
+    dtype : npt.DTypeLike
+        Data type of the array.
+    chunk_shape : ChunkCoords
+        Chunk shape of the array.
+    shard_shape : ChunkCoords, optional
+        Shard shape of the array. The default value of ``None`` results in no sharding at all.
+    filters : Iterable[Codec], optional
+        List of filters to apply to the array.
+    compression : Iterable[Codec], optional
+        List of compressors to apply to the array.
```

Comment:
These definitions could use some tuning. Should we advertise `filters` as `array->array` codecs and `compressors` as `bytes->bytes` codecs? Some further constraints here seem important to helping us decide on behavior and users to understand how to configure them.

Reply:
agreed, I will add some clarifying language